### ISSUE TITLE
Missing git status color indicator in tree view 

### ISSUE THREAD

**Issue by kcmr** [Microsoft/vscode#46888]
Issue Type: <b>Bug</b>

1. Open a project under VCS (git)
2. Make some changes
3. The modified files don't appear with the corresponding colors in the tree view until a new file is created.

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:39:09.335Z)
OS version: Darwin x64 17.4.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz (4 x 2700)|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (4.11GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (59)</summary>

Extension|Author (truncated)|Version
---|---|---
All-Autocomplete|Ati|0.0.11
vscode-toggle-quotes|Bri|0.2.0
column-jump|Coe|0.0.4
EditorConfig|Edi|0.12.1
vsc-material-theme|Equ|1.5.1
todo-tree|Gru|0.0.31
beautify|Hoo|1.3.0
material-icon-theme|PKi|3.2.6
ejs-language-support|Qas|0.0.1
vscode-paste-and-indent|Rub|0.0.8
js-complexity-analysis|Tom|2.0.0
vscode-color|ans|0.4.5
halcyon-vscode|bri|0.0.9
simple-react-snippets|bur|1.1.1
polymer-snippets|chi|1.0.1
vscode-eslint|dba|1.4.8
vscode-instant-markdown|dba|1.4.1
vscode-quick-select|dba|0.2.5
vscode-babel-coloring|dza|0.0.4
eppz-code|epp|1.2.41
json-tools|eri|1.0.2
file-icons|fil|1.0.12
vscode-flatland-icon-theme|geb|0.0.6
open-in-editor-vscode|gen|0.9.7
bracketselection|guo|1.0.3
githd|hui|1.1.3
svg|joc|0.0.9
theme-karyfoundation-themes|kar|17.0.0
vscode-html-format|moh|0.0.1
sublime-keybindings|ms-|3.0.3
debugger-for-chrome|msj|4.2.2
vscode-open-iterm2|mus|0.0.4
color-highlight|nau|2.3.0
prettier-vscode-with-tabs|pas|1.1.0
vscode-code-outline|pat|0.0.13
vscode-template-literal-editor|pli|0.8.4
polacode|pnp|0.2.2
polymer-ide|pol|0.6.0
subtle-brackets|raf|2.1.0
nunjucks|ron|0.2.3
vscode-idiomatic-css-comments|rya|0.1.2
vscode-coverage-gutters|rya|1.3.1
bracket-jumper|sas|1.1.8
vscode-table-formatter|shu|1.2.1
bunker|sld|1.1.6
vscode-fileutils|sle|2.8.1
Align|ste|0.2.0
vscode-field-lights|sve|0.0.8
ayu|tea|0.14.0
open-in-browser|tec|1.1.0
colonizer|uri|0.8.0
expand-selection-to-scope|vit|0.2.0
gitblame|wad|2.4.0
vscode-js-console-utils|wht|0.7.0
change-case|wma|1.0.0
better-align|wwm|1.1.6
JavaScriptSnippets|xab|1.5.0
markdown-all-in-one|yzh|1.1.1
editorzoom|zol|1.0.3

(97 theme extensions excluded)

</details>
Reproduces without extensions

**Comment by jrieken** [comment]
Must be more to it than just that. Can you share the repo and the file in which this happens

**Comment by kcmr** [comment]
Sorry, I can't share the repo because it's private. But I think that the project has nothing to do with the issue because it happens in all my projects under git.

It started to happen after installing and uninstalling this extension: [AB HTML Formatter](https://marketplace.visualstudio.com/items?itemName=zovorap.ab-html-formatter)

This is a screen recording of the behavior: https://cl.ly/qWCq

**Comment by jrieken** [comment]
Can you share the git log (View > Output > Git) with us? This feature is used extensively so I doubt it is generally broken but we must be hitting a special case here

**Comment by kcmr** [comment]
Yes, here you go:

```
Buscando git en: /usr/bin/git
Usando GIT 2.14.3 (Apple Git-98) desde /usr/bin/git
> git rev-parse --show-toplevel
> git config --get commit.template
Open repository: /Users/maria.camara/projects/github/color-your-code
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name --abbrev-ref master@{u}
> git rev-list --left-right master...origin/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git check-ignore -z --stdin
> git show :src/cyc-app/cyc-app.css
> git show :src/cyc-app/new-file
> git check-ignore -z --stdin
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name --abbrev-ref master@{u}
> git rev-list --left-right master...origin/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git show :src/cyc-app/cyc-app.css
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name --abbrev-ref master@{u}
> git rev-list --left-right master...origin/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git check-ignore -z --stdin
> git show :src/cyc-app/cyc-app.html
> git show :.editorconfig
> git show :.gitignore
> git show :electron-color-theme.json
```

**Comment by jrieken** [comment]
Hm, the command for ignored files doesn't seem to complain (`> git check-ignore -z --stdin`).. You could also check the extension host log files (F1 > Show Log... > Extension Host) and look for something suspicious. 

Then, wrt to modified files. How does the git viewlet update? Is that instant as you make changes? Both models should update in sync, when git status is executed...

**Comment by kcmr** [comment]
Sorry, I've just restarted my computer and everything works fine now without the need to make a change.

Until now, the tree view was updated only after creating a new file. I'm not sure if you are asking me about that.

This is the Extension Host:

```
[2018-03-29 11:35:21.123] [exthost1] [info] extension host started
[2018-03-29 11:35:21.165] [exthost1] [info] ExtensionService#_doActivateExtension vscode.emmet {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.165] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/emmet/out/extension
[2018-03-29 11:35:21.241] [exthost1] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.241] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/git/out/main
[2018-03-29 11:35:21.401] [exthost1] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.401] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/merge-conflict/out/extension
[2018-03-29 11:35:21.426] [exthost1] [info] ExtensionService#_doActivateExtension Atishay-Jain.All-Autocomplete {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.426] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/Atishay-Jain.all-autocomplete-0.0.11/out/src/extension
[2018-03-29 11:35:21.454] [exthost1] [info] ExtensionService#_doActivateExtension EditorConfig.EditorConfig {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.454] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/EditorConfig.editorconfig-0.12.1/out/src/editorConfigMain
[2018-03-29 11:35:21.500] [exthost1] [info] ExtensionService#_doActivateExtension Equinusocio.vsc-material-theme {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.500] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/Equinusocio.vsc-material-theme-1.5.1/extensions/material.theme.config.js
[2018-03-29 11:35:21.516] [exthost1] [info] ExtensionService#_doActivateExtension Gruntfuggly.todo-tree {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.516] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/Gruntfuggly.todo-tree-0.0.31/extension
[2018-03-29 11:35:21.523] [exthost1] [info] ExtensionService#_doActivateExtension HookyQR.beautify {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.523] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/HookyQR.beautify-1.3.0/extension
[2018-03-29 11:35:21.597] [exthost1] [info] ExtensionService#_doActivateExtension PKief.material-icon-theme {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.597] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/PKief.material-icon-theme-3.2.6/out/src/extension
[2018-03-29 11:35:21.665] [exthost1] [info] ExtensionService#_doActivateExtension dbaeumer.vscode-eslint {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.665] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/dbaeumer.vscode-eslint-1.4.8/client/out/extension
[2018-03-29 11:35:21.743] [exthost1] [info] ExtensionService#_doActivateExtension eppz.eppz-code {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:21.743] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/eppz.eppz-code-1.2.41/out/src/main
[2018-03-29 11:35:22.028] [exthost1] [info] ExtensionService#_doActivateExtension huizhou.githd {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.028] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/huizhou.githd-1.1.3/out/src/extension
[2018-03-29 11:35:22.044] [exthost1] [info] ExtensionService#_doActivateExtension ms-vscode.sublime-keybindings {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.044] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/ms-vscode.sublime-keybindings-3.0.3/extension
[2018-03-29 11:35:22.045] [exthost1] [info] ExtensionService#_doActivateExtension naumovs.color-highlight {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.045] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/naumovs.color-highlight-2.3.0/dist/extension
[2018-03-29 11:35:22.083] [exthost1] [info] ExtensionService#_doActivateExtension rafamel.subtle-brackets {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.083] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/rafamel.subtle-brackets-2.1.0/out/src/extension
[2018-03-29 11:35:22.218] [exthost1] [info] ExtensionService#_doActivateExtension ryanluker.vscode-coverage-gutters {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.218] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/ryanluker.vscode-coverage-gutters-1.3.1/out/src/extension
[2018-03-29 11:35:22.543] [exthost1] [info] ExtensionService#_doActivateExtension waderyan.gitblame {"startup":true,"activationEvent":"*"}
[2018-03-29 11:35:22.543] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/waderyan.gitblame-2.4.0/out/src/index
[2018-03-29 11:35:22.582] [exthost1] [info] ExtensionService#_doActivateExtension vscode.markdown-language-features {"startup":true,"activationEvent":"workspaceContains:README.md"}
[2018-03-29 11:35:22.582] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/markdown/out/extension
[2018-03-29 11:35:22.706] [exthost1] [info] eager extensions activated
[2018-03-29 11:35:23.058] [exthost1] [info] ExtensionService#_doActivateExtension vscode.configuration-editing {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2018-03-29 11:35:23.058] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/configuration-editing/out/extension
[2018-03-29 11:35:23.074] [exthost1] [info] ExtensionService#_doActivateExtension vscode.json {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2018-03-29 11:35:23.074] [exthost1] [info] ExtensionService#loadCommonJSModule /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/json/client/out/jsonMain
[2018-03-29 11:35:23.556] [exthost1] [info] ExtensionService#_doActivateExtension yzhang.markdown-all-in-one {"startup":true,"activationEvent":"workspaceContains:README.md"}
[2018-03-29 11:35:23.556] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/yzhang.markdown-all-in-one-1.1.1/out/src/extension
[2018-03-29 11:35:35.672] [exthost1] [info] ExtensionService#_doActivateExtension polymer.polymer-ide {"startup":false,"activationEvent":"onLanguage:plaintext"}
[2018-03-29 11:35:35.672] [exthost1] [info] ExtensionService#loadCommonJSModule /Users/maria.camara/.vscode/extensions/polymer.polymer-ide-0.6.0/out/src/extension
```

We can close the issue until I can reproduce it again.

**Comment by jrieken** [comment]
Ok - please keep an eye open. Thanks for the responses.

**Comment by crichmond1989** [comment]
Hey Team,

I'm having the exact same issue. Can't tell if I should just comment here, or write a new issue.

Issue Type: <b>Bug</b>

File explorer missing git status color

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:47:13.351Z)
OS version: Windows_NT ia32 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz (8 x 3500)|
|Memory (System)|7.94GB (3.73GB free)|
|Process Argv|C:\Program Files (x86)\Microsoft VS Code\Code.exe e:\git\deltaleague|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-html-css|ecm|0.2.0
code-runner|for|0.9.3
dotenv|mik|1.0.1
csharp|ms-|1.14.0
PowerShell|ms-|1.6.0
vscode-docker|Pet|0.0.26
jenkinsfile-support|sec|0.1.0


</details>
Reproduces without extensions

**Comment by crichmond1989** [comment]
@jrieken I updated VS Code on my Ubuntu machine and have mostly the same issue as my Windows 10 machine. The "modified" status color and icon only shows up for the top-most folder. The files and sub-directories are plain. Interestingly, the files and sub-directories have the correct "* Modified" tool-tip.

Let me know if you need more details.

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:46:30.761Z)
OS version: Linux x64 4.13.0-37-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (8 x 2600)|
|Load (avg)|1, 1, 1|
|Memory (System)|7.65GB (1.00GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
beautify|Hoo|1.3.0
vscode-docker|Pet|0.0.26
gitlens|eam|8.2.0


</details>
Reproduces without extensions
